.github/workflows/ci-validation.yml

```diff
@@ -35,9 +35,15 @@ jobs:
     - name: Build (Ubuntu 20.04)
       run: |
         sudo cp -rf . /p4c/
-        (cd /p4c/ && sudo -E tools/ci-build.sh)
+        (cd /p4c/ && sudo -E tools/ci-build.sh) || echo "BUILD_SUCCESS=false" >> $GITHUB_ENV
         sudo cp -rf /p4c/.ccache .
 
     - name: Validate
-      run: sudo -E ctest  -R toz3-validate-p4c --output-on-failure --schedule-random
-      working-directory: /p4c/build
+      if: env.BUILD_SUCCESS == 'true'
+      run: |
+          ctest  -R toz3-validate-p4c --output-on-failure --schedule-random
```

Comment:
Will it run? dont you need sudo -E?

i am suprised why you need any changes at all in this file

Reply:
Some hours ago when I was looking why validate fails it was due to some leftover of removed build_unified.

Reply:
Try to avoid any changes here and see what CI says

Reply:
Yes it should run without sudo. `sudo` is only required for eBPF kernel tests.  These changes need to be preserved to avoid build failures of Gauntlet affecting PRs. This is because Gauntlet is a downstream dependency.

Reply:
But this is still broken. Validate Step ends in Build failed (but build went fine, test image was created so strange…. what is worse, it is green, so easy to overlook any issues - I changed it in my previous PR but I am not sure why you reverted those changes).

Reply:
Yes, that is intentional. Failed builds should not show up in PRs since they can only be fixed in the downstream repository. 

I will fix the odd skipping.